Woof/Woof/ViewModel/SitterProfileViewModel.swift

```diff
@@ -46,6 +50,22 @@ final class SitterProfileViewModel: ObservableObject {
             .save(data, for: KeyValueStorage.Key.currentSitter)
     }
 
+    /// Uploads the pet sitter's data to the backend server asynchronously.
+    ///
+    /// `isErrorOccurred` is set to `false`, indicating no errors.
+    /// If there is a network error during the upload,`isErrorOccurred` is set to `true`.
+    func upload() async throws {
+        do {
+            let endpoint = WoofAppEndpoint.addNewSitter(currentSitter.asDictionary())
+            _ = try await networkService.request(endpoint)
+        } catch {
+            await MainActor.run {
+                isErrorOccurred = true
+            }
+            throw error
```

Comment:
what this line does? Do we need it?

Reply:
I don't think that is a good idea and I don't really understand why we should throw an error if it is the last point where we have to handle the error.

Reply:
Fixed